File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int elem = nums[0];
        int cnt = 1;
        for(int i = 1; i < nums.size(); i++){
            if(cnt == 0){
                elem = nums[i];
                cnt = 1;
            }
            else if(nums[i] == elem) cnt++;
            else if(nums[i] != elem) cnt--;
        }
        return elem;
    }
};

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        cnt = 1
        elem = nums[0]
        for i in range(1, len(nums)):
            if(nums[i] == elem):
                cnt += 1
            elif(nums[i] != elem):
                cnt -= 1
                if(cnt == 0):
                    elem = nums[i]
                    cnt = 1
        return elem  